src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+type NumberValue = number;
+
+export class Triangle implements Figure {
+  readonly shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: NumberValue = 0,
+    public b: NumberValue = 0,
+    public c: NumberValue = 0,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid triangle dimensions');
```

Comment:
[CODE STYLE] - If you throw an error, try to provide a detailed description in the error message. For instance, 'Invalid triangle dimensions' could be more descriptive by specifying that all sides must be greater than zero.

Reply:
> [CODE STYLE] - If you throw an error, try to provide a detailed description in the error message. For instance, 'Invalid triangle dimensions' could be more descriptive by specifying that all sides must be greater than zero.

Good point, it's better to provide more details about the reason why error occured